pkg/cacheutil/memcached_client.go

```diff
@@ -467,15 +467,20 @@ func (c *memcachedClient) getMultiBatched(ctx context.Context, keys []string) ([
 	// Allocate a channel to store results for each batch request. The max concurrency will be
 	// enforced by doWithBatch.
 	results := make(chan *memcachedGetMultiResult, numResults)
-	defer close(results)
+	// NOTE: we are not closing the results channel here on purpose. doWithBatch will start
```

Comment:
I think I've addressed your concerns @bwplotka:

* I've added back the code you mentioned
* I've switched the call to `doWithBatch` to use a background context to ensure it executes the expected number of batches even when the `ctx` passed to `getMultiBatch` is canceled. The check of the context in `getMultiSingle` is retained so that we can still avoid doing work when the context is cancelled.